### PR TITLE
BUGFIX: SD-926 add Mapbox filter by using attribute uri

### PR DESCRIPTION
**Pull request by naibinh**
Root cause:
- Masked attribute items have same masking value (eg `Undisclosed`) which is not accepted by mapbox filter.

Solution:
- Replace attribute value/name to its uri which is more unique

---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: [PR](https://github.com/gooddata/gdc-analytical-designer/pull/3112), [client-demo](https://client-demo.na.intgdc.com:50033/analyze/)
- gdc-dashboards: [PR](https://github.com/gooddata/gdc-dashboards/pull/2828), [client-demo](https://client-demo.na.intgdc.com:50035/dashboards/)

# Related Jira tasks
- SD-926: https://jira.intgdc.com/browse/SD-926

### CODE REVIEW

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7435/ : SUCCESS in 3m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3456/ : SUCCESS in 3s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3879/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3457/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3231/ : SUCCESS in 10m 20s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1612.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3458/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3232/ : SUCCESS in 16m 46s

File: src/helpers/geoChart/data.ts [hunk 1]
```diff
@@ -64,6 +65,7 @@ export function getGeoData(
 
     if (segmentIndex !== undefined) {
         geoData[SEGMENT].data = getAttributeData(attributeHeaderItems, segmentIndex);
+        geoData[SEGMENT].uris = getAttributeUris(attributeHeaderItems, segmentIndex);
```
Comment (ptquang86) [inline src/helpers/geoChart/data.ts:68]: do you want to merge these 2 lines into 1?

Comment (naibinh) [reply src/helpers/geoChart/data.ts:68]: Do you mean?
```
geoData[SEGMENT] = {
    ...geoData[SEGMENT],
    data: getAttributeData(...), 
    uris: getAttributeUris(...)
}
```
If so, I would prefer the first style

Comment (ptquang86) [reply src/helpers/geoChart/data.ts:68]: I meant: we will create a new functon to get all data and uris for segment. 
Use only 1 loop to save time

Comment (naibinh) [reply src/helpers/geoChart/data.ts]: Fixed. Thanks

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7447/ : SUCCESS in 2m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3480/ : SUCCESS in 3s

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7452/ : SUCCESS in 3m 34s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3484/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3898/ : SUCCESS in 12m 47s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/36/ : SUCCESS in 18s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7471/ : SUCCESS in 2m 46s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1811/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2665/ : SUCCESS in 20m 34s

**Pull request merged**